### ISSUE TITLE
UploadFileToBlockBlob: Status: 400 The value for one of the HTTP headers is not in the correct format.

### ISSUE THREAD

**Issue by jbmcfarlin31** [Azure/azure-storage-blob-go#176]
### Which version of the SDK was used?
`v0.8.0`

### Which platform are you using? (ex: Windows, Linux, Debian)
`Linux - Ubuntu 16.04`

### What problem was encountered?
I was trying to upload a file to my blob container in AzureStack and it returned an error saying `400 The value for one of the HTTP headers is not in the correct format`. I was following the [example code here](https://github.com/Azure/azure-storage-blob-go/blob/master/azblob/zt_examples_test.go#L1061).

Here is the code I was using:
```Go
// From the Azure portal, get your Storage account blob service URL endpoint.
accountName, accountKey := accountInfo()

// Create a BlockBlobURL object to a blob in the container (we assume the container already exists).
u, _ := url.Parse(fmt.Sprintf("https://%s.blob.my.domain.com/testcontainer/%s", accountName, fileName))

credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)

if err != nil {
	log.Fatal(err)
}

blockBlobURL := azblob.NewBlockBlobURL(*u, azblob.NewPipeline(credential, azblob.PipelineOptions{}))

ctx := context.Background() // This example uses a never-expiring context

fileVal, err := os.Open(fileName)
if err != nil {
	log.Fatal(err)
}

defer fileVal.Close()

fileSize, err := fileVal.Stat() // Get the size of the file (stream)
if err != nil {
	log.Fatal(err)
}

response, err := azblob.UploadFileToBlockBlob(ctx, fileVal, blockBlobURL,
	azblob.UploadToBlockBlobOptions{
		Progress: func(bytesTransferred int64) {
			fmt.Printf("Uploaded %d of %d bytes.\n", bytesTransferred, fileSize.Size())
		},
	})

// Verify that upload was successful
if err != nil {
	log.Fatal(err)
}

_ = response
```

It returned with the following:
```
2020/05/27 22:02:13 -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /root/odev/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.8.0/azblob/zc_storage_error.go:42
===== RESPONSE ERROR (ServiceCode=) =====
Description=The value for one of the HTTP headers is not in the correct format.
RequestId:be906bf1-9478-0384-7651-2fc183bbf148
Time:2020-05-27T22:02:32.3229462Z, Details:
   Code: InvalidHeaderValue
   HeaderName: x-ms-version
   HeaderValue: 2018-11-09
   PUT https://REDACTED/testcontainer/README.md?timeout=61
   Authorization: REDACTED
   Content-Length: [114]
   User-Agent: [Azure-Storage/0.7 (go1.13.10; linux)]
   X-Ms-Blob-Cache-Control: []
   X-Ms-Blob-Content-Disposition: []
   X-Ms-Blob-Content-Encoding: []
   X-Ms-Blob-Content-Language: []
   X-Ms-Blob-Content-Type: []
   X-Ms-Blob-Type: [BlockBlob]
   X-Ms-Client-Request-Id: [f2ad1907-d35c-4088-4020-1fbca7ab44a5]
   X-Ms-Date: [Wed, 27 May 2020 22:02:13 GMT]
   X-Ms-Version: [2018-11-09]
   --------------------------------------------------------------------------------
   RESPONSE Status: 400 The value for one of the HTTP headers is not in the correct format.
   Content-Length: [328]
   Content-Type: [application/xml]
   Date: [Wed, 27 May 2020 22:02:32 GMT]
   Server: [Microsoft-HTTPAPI/2.0]
   X-Ms-Request-Id: [be906bf1-9478-0384-7651-2fc183bbf148]
```
### How can we reproduce the problem in the simplest way?
Pull the snippet of code from the link above, lines 1037-1071 and run it.

### Have you found a mitigation/solution?
No, not yet.

**Comment by jbmcfarlin31** [comment]
Update:
It looks like this might be an issue with the API profile version being used. As seen above, the profile version is `2018-11-09`. When I inspect from within the Azure Stack portal, it looks like the request headers being passed through for the same operation are similar however the version in that case is: `2017-04-17`.

I think this might be solved by using `v0.2.0` of the SDK, so I am attempting to try that. I will update my findings as I get them.

**Comment by jbmcfarlin31** [comment]
Update #2:
This was an issue with the API version for my particular case. I had to follow the import path and version reference located [here](https://github.com/Azure/azure-storage-blob-go/). I also had to look at the current API profiles for the latest version of Azure Stack - located [here](https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-acs-differences?view=azs-2002).

With that being said, in order to import `v0.2.0` of this SDK, you have to omit the `v` in the version:
```bash
go get github.com/Azure/azure-storage-blob-go/azblob@0.2.0
```
This allowed me to get what I needed, referencing the import:
```Go
import "github.com/Azure/azure-storage-blob-go/2017-07-29/azblob"
```
I do expect that not all functionality and latest updates will be in this release, however for what I needed it to do (copy files up to a blob container), it worked perfectly.
